x/mongo/driver/topology/topology.go

```diff
@@ -321,13 +347,33 @@ func (t *Topology) Connect() error {
 	}
 
 	t.serversLock.Unlock()
-	if t.pollingRequired {
-		uri, err := url.Parse(t.cfg.URI)
-		if err != nil {
-			return err
+	uri, err := url.Parse(t.cfg.URI)
+	if err != nil {
+		return err
+	}
+	parsedHosts := strings.Split(uri.Host, ",")
+	if mustLogTopologyMessage(t, logger.LevelInfo) {
```

Comment:
Optional: the only purpose of this block is to log third party connection information. It could be cleaner to create a separate function called `logTopologyThirdPartyUsage` to hold all of this logic:

```go
if mustLogTopologyMessage(t, logger.LevelInfo) {
    logTopologyThirdPartyUsage(...)
}
```